### PR TITLE
feat: Add `KraftKit` contributing page

### PR DESCRIPTION
**Pull request by LucaSeri**
The `KraftKit` contributing page has the same format as the `Unikraft core` contributing page, which has been modified to reflect the conventions used in the `KraftKit` development.

GitHub-fixes: https://github.com/unikraft/docs/issues/370

### CODE REVIEW

**Review by StefanJum — CHANGES_REQUESTED** [review]
@LucaSeri you need to update the sidebar for this page to show up. See [this](https://github.com/unikraft/docs/pull/378/files#diff-ccba73597f02fa542023f0fd7342334cd299037dc3a42abc41fa4bab336342a0) for reference.

LE: nvm, the entry is already there, sorry.

File: content/docs/contributing/kraftkit.mdx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+## Rebasing and Squashing
+
+While working on a pull request, the destination branch may change.
+This will require [rebasing](https://docs.github.com/en/get-started/using-git/about-git-rebase) the source branch to keep it in sync.
+There may be conflicts that need to be solved as part of the rebase.
+
+As you want to create on commit for each content change, [squashing multiple commits](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/managing-commits/squashing-commits) into one may be required.
+This may be either because of different small changes added to the pull request as new commits.
+Or due to [incorporating suggested changes](docs/contributing/suggest-changes).
```
Comment (StefanJum) [inline content/docs/contributing/kraftkit.mdx:147]:
```suggestion
Or due to [incorporating suggested changes](/docs/contributing/suggest-changes).
```

File: content/docs/contributing/kraftkit.mdx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+The Unikraft project is an open source and encourages the fostering open
+collaboration. If you are reading this guide on how to submit changes, then
+thank you in advance!
+
+Unikraft's source code is hosted on [GitHub](https://github.com). This includes
+the Unikraft core repository, auxiliary external microlibraries, applications,
+platforms, architectures and additional tools like `kraft`. Any of these
+repositories follow the same submission process for changes: the pull request.
+
+Make sure one pull request covers only one topic.
```
Comment (StefanJum) [inline content/docs/contributing/kraftkit.mdx:17]: Use one sentence per line and one line per sentence here.

File: content/docs/contributing/kraftkit.mdx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+### Submission Checklist
+
+When you make a submission, please make sure you follow the steps below, also
+part of the pull request template:
```
Comment (StefanJum) [inline content/docs/contributing/kraftkit.mdx:22]: Same, one line per sentence.

File: content/docs/contributing/kraftkit.mdx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+Make sure each commit corresponds to one code / content change only.
+
+Please create descriptive commit messages. Consider using a prefix for the
+commit message. Add a detailed description on the motivation for the commit and
+summary of changes. Follow [this guide](https://cbea.ms/git-commit/) on writing
+good commit messages.
+
+Each commit must be authored by adding a `Signed-off-by` message.  Use the
+`-s|--signoff` option of the `git` command when creating a commit.
```
Comment (StefanJum) [inline content/docs/contributing/kraftkit.mdx:41]: Same as above.

**Review by StefanJum — CHANGES_REQUESTED** [review]
See the comments. Also add a link somewhere to some `go` coding conventions, I assume there must be a list of what `go fmt` checks for somewhere.

**Comment by LucaSeri** [comment]
Thanks for the review @StefanJum, I added your suggested changes.
>  Also add a link somewhere to some `go` coding conventions, I assume there must be a list of what `go fmt` checks for somewhere.
I linked ``gofumpt``, their readme provides a bunch of examples about the rules, please check if you think that's enough.

**Review by StefanJum — APPROVED** [review]
> I linked gofumpt, their readme provides a bunch of examples about the rules, please check if you think that's enough.

It's perfect.

All good now, thanks.

Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by razvand — APPROVED** [review]
Thanks, @LucaSeri 

Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**